Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -113,6 +113,84 @@ class FileLogSource : LogSource {
     }
 }
 
+class EventLogLogSource : LogSource {
+    [String] $EventLogName
+    [String] $EventLogSource
+    [int64] $StartEventIdx
+
+    EventLogLogSource($Session, $EventLogName, $EventLogSource) {
+        $This.Session = $Session
+        $this.EventLogName = $EventLogName
+        $this.EventLogSource = $EventLogSource
+        $this.StartEventIdx = $this.GetLatestEventIdx() + 1
+    }
+
+    [CollectedLog[]] GetContent() {
+        $LogGetterBody = {
+            Param([Parameter(Mandatory = $true)] [string] $LogName,
+                  [Parameter(Mandatory = $true)] [string] $LogSource,
+                  [Parameter(Mandatory = $true)] [int64] $StartEventIdx,
+                  [Parameter(Mandatory = $true)] [int64] $EndEventIdx)
+
+            $Name = "$LogSource - $LogName event log"
+
+            try {
+                $Content = Get-EventLog `
+                    -LogName $LogName `
+                    -Source $LogSource `
+                    -Index ($StartEventIdx..$EndEventIdx) | Format-Table -Wrap | Out-String
```

Comment:
If number of messages in event log hadn't changed from the moment of initialization, this would produce wrong outcome. 
Let's say there are 10 messages in event log, then at the moment of creation StartEventIdx = 10 + 1 = 11.  Then we invoke GetContent() (when there are still 10 messages in event log), so StartEventIdx =11 and EndEventIdx = 10. (11..10) gets 10th message from event log and we don't want that.